app/models/government.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+class Government
+  include ActiveModel::Model
+
+  def self.find(id)
+    all.find { |government| government.id == id }
+  end
+
+  def self.for_date(date)
+    all.find { |government| government.covers?(date) }
+  end
+
+  def self.current
+    for_date(Date.current)
+  end
+
+  def self.all
+    @all ||= YAML.load_file(Rails.root.join("config", "governments.yml"))
+                 .map { |hash| new(hash) }
```

Comment:
Sure, the `new` is the equivalent of calling `Government.new` since we're in the scope of the class.

So what this code is doing is looping through each object in the YAML file and creating an instance of Government for each one.

Reply:
Ah ok - I would normally just explicitly write `Government.new`, which removes any ambiguity, but use of `new` on its own is probably just a Ruby idiom I haven't come across before. I appreciate it means one less thing to update if you ever have to rename the class.

Reply:
Think of it in the same way that when we call instance methods we don't prefix them with `self` in the same way we prefix with `this` in Javascript.

In this situation we're in the context of Government so self == Government. So this method could be `self.new`, `Government.new` or `new`. Convention in ruby would dictate not using a superfluous prefix.